test/index.js

```diff
@@ -1313,3 +1396,933 @@ test('contramap', function() {
 
   eq(Z.contramap(length, inc)('abc'), 4);
 });
+
+function testLaws(specs) {
+  Object.keys(specs).forEach(function(typeClassName) {
+    suite(typeClassName, function() {
+      var spec = specs[typeClassName];
```

Comment:
I like this approach! I might refactor https://github.com/fluture-js/Fluture/pull/220 to use a similar one!
Also, it looks like if you would do `var module = laws[typeClassName]`, you wouldn't need to repeat yourself.

```diff
-testLaws({Setoid: {module: laws.Setoid, laws: {reflexivity: []}}})
+testLaws({Setoid: {reflexivity: []}})
```

Reply:
Ah, I scrolled down to the part where modules require arguments. I understand now.

Reply:
I just scrolled to the part where module require arguments. I understand now.

Reply:
I'm pleased you like this approach. I spent many hours trying different approaches, on several occasions making the code less repetitive only to change my mind and remove the indirection.

At one point I had this:

```javascript
suite('Boolean', function() {
  var arb = jsc.bool;
  testLaws(Z.equals, {
    Setoid: {
      'Setoid a => Arbitrary a': arb
    },
    Ord: {
      'Ord a => Arbitrary a': arb
    }
  });
});
```

An object provided mappings from these descriptively labelled values to positional arguments. A given arbitrary (e.g. `Setoid a => Arbitrary a`) could be used to test several of a type class's laws, and could be used multiple times when testing certain laws (e.g. associativity).

The approach did not work well, though, for more complex laws. The Traversable laws proved particularly troublesome:

```javascript
Traversable: {
  'Traversable t => Arbitrary (TypeRep t)': jsc.constant(List),
  'Traversable u => Arbitrary (TypeRep u)': jsc.constant(Array),
  'Traversable t => Arbitrary (t a)': arb,
  '(Applicative f, Traversable t, Traversable u) => Arbitrary (f (t a) -> f (u a))': jsc.constant(listToArray),
  '(Traversable t, Traversable u) => Arbitrary (u (t a))': jsc.array(ListArb(jsc.number)),
  'Applicative f => Arbitrary (TypeRep f)': jsc.constant(Maybe),
  'Applicative g => Arbitrary (TypeRep g)': jsc.constant(List),
  '(Applicative f, Applicative g, Traversable t) => Arbitrary (t (f (g a)))': jsc.array(MaybeArb(ListArb(jsc.number)))
}
```

The relationships between the values above are not obvious. A more direct use of fantasy-laws is clearer, and I can live with the repetition.